svgdigitizer/electrochemistry/cv.py

```diff
@@ -18,111 +18,166 @@
         units given in the figure description.
         '''
-        df_ = df.copy()
-        if self.yunit == 'A':
-            df_['I_A'] = df['I']
 
-        if self.yunit == 'mA':
-            df_['I_A'] = df['I']/1E3
+        df_ = df.copy()
+        q = 1 * self.get_axis_unit('y')
 
-        if self.yunit == 'uA':
-            df_['I_A'] = df['I']/1E6
+        # Verify if the y data is current ('A') or current density ('A / cm2')
+        if 'm2' in str(q.unit):
+            conversion_factor = q.to(u.A / u.m**2)
+        else:
+            conversion_factor = q.to(u.A)
 
-        df_['I_mA'] = df_['I_A']*1E3
-        df_['I_uA'] = df_['I_A']*1E6
+        df_[self.axis_properties['y']['dimension']] = df_['y'] * conversion_factor
 
-        return df_[['I_A', 'I_mA', 'I_uA']]
+        return df_[[self.axis_properties['y']['dimension']]]
 
     def create_df_time_axis(self, df):
         r'''
         Create a time axis in the dataframe based on the scan rate given in the
         figure description.
         '''
-        df_ = df.copy()
-        df_['deltaU'] = abs(df_[self.xlabel].diff())
-        df_['cumdeltaU'] = df_['deltaU'].cumsum()
-        df_['t'] = df_['cumdeltaU']/self.get_rate()
-        return df_['t']
+        df = df.copy()
+        df['deltaU'] = abs(df['U'].diff().fillna(0))
+        df['cumdeltaU'] = df['deltaU'].cumsum()
+        df['t'] = df['cumdeltaU']/float(self.rate.to(u.V / u.s).value)
+
+        return df[['t']]
 
     def plot_cv(self):
-        self.df.plot(x='U_V', y='I_uA')
-        plt.xlabel(f'{self.xlabel} / {self.xunit}')
-        plt.ylabel(f'{self.ylabel} / {self.yunit}')
+        self.cv_df.plot(x=self.axis_properties['x']['dimension'], y=self.axis_properties['y']['dimension'])
+        plt.axhline(linewidth=1, linestyle=':', alpha=0.5)
+        plt.xlabel(self.axis_properties['x']['dimension'] + ' / ' + str(self.axis_properties['x']['unit']))
+        plt.ylabel(self.axis_properties['y']['dimension'] + ' / ' + str(self.axis_properties['y']['unit']))
+
+    @property
+    def metadata_out(self, comment=''):
+        # Add description
+        assert type(comment) == str, 'Comment must be a string'
+        metadata = self.metadata
+        if 'figure description' not in self.metadata:
+            metadata['figure description'] = {}
+
+        metadata['figure description']['type'] = 'digitized'
+        metadata['figure description']['measurement type'] = 'CV'
+        metadata['figure description']['scan rate'] = {'value': self.rate.value, 'unit': str(self.rate.unit)}
+        if 'potential scale' not in metadata['figure description']:
+            metadata['figure description']['potential scale'] = {}
+        metadata['figure description']['potential scale']['unit'] = str(self.get_axis_unit('x'))
+        # Implement: Get the reference from the text labels of the axis
+        # metadata['figure description']['potential scale']['reference'] = get_from_ax_labels
```

Comment:
Is there an issue for this? We should write `TODO: ... #ABC` so it is clear that something is missing here.
There shuold be no commented-out code. It's afterwards almost impossible to understand why it was commented out ;)